openedx/core/djangoapps/user_authn/exceptions.py

```diff
@@ -20,3 +21,11 @@ def get_response(self):
                 resp[attr] = self.__getattribute__(attr)
 
         return resp
+
+    def sanitize_value(self, value):
+        """ Sanitize error message for safe embed on login page."""
+        output = bleach.clean(
+            value, tags=['a', 'p', 'br', 'strong', 'b', 'span']
+        )
```

Comment:
@robrap Is this the recommended approach for sanitizing?

Reply:
It depends.  Some questions:
1. What is generating errors that already contain HTML?  If the error must contain pre-existing HTML, bleaching it is wise and correct.
2. The biggest watch-out here is with the anchor tags.  I don't know for sure, but I don't think bleach ensures that the URL for an anchor tags is safe, so it depends where that URL is coming from.  Is the URL potentially user supplied (i.e. through a query param)?  If so, could someone supplied a bad URL or javascript, we would not want to display it back to the user.

Reply:
@robrap Yes, the generating errors already contain HTML, for example see https://github.com/edx/edx-platform/blob/07696ef5dd45af6c22a0ce17defc38d28c8dc1d1/openedx/core/djangoapps/user_authn/views/login.py#L144-L147

For your second question, URL for anchor tag is our internal URL (i.e. support url from settings), so I think it's not an issue either.

Reply:
I see the message now. Given it is translated, it would be better to use the Text/HTML functions documented in readthedocs, where the HTML chunks are converted to format strings that are interpolated in (eg start_link, end_link, etc.). 

The call here where you are setting value and using sanitize function could instead call Text again, which should be a no-op if the message is already set using Text.

Writing this from phone. Let me know if you have any questions.

Reply:
@robrap So, you're suggesting to use Text/HTML functions where these messages contains HTML and also call Text() on value in exception class where I'm sanitizing to handle the other cases where we only have text, make sense.

I'll let you know after updating the PR, thanks.

Reply:
Yes.  And this may already have been clear, but sanitizing will go away.  We should really use sanitizing for HTML that is outside of our control, like user input as an example.